src/Tools/AI Test Toolkit/src/TestSuite/AITALTestSuiteMgt.Codeunit.al

```diff
@@ -0,0 +1,202 @@
+    end;
+
+    internal procedure ExpandCodeunit(var AITTestMethodLine: Record "AIT Test Method Line")
+    var
+        TestInput: Record "Test Input";
+    begin
+        TestInput.SetRange("Test Input Group Code", AITTestMethodLine.GetTestInputCode());
+        TestInput.ReadIsolation := TestInput.ReadIsolation::ReadUncommitted;
+        if not TestInput.FindSet() then
+            exit;
+
+        repeat
+            this.ExpandCodeunit(AITTestMethodLine, TestInput);
+        until TestInput.Next() = 0;
+    end;
+
+    internal procedure ExpandCodeunit(var AITTestMethodLine: Record "AIT Test Method Line"; var TestInput: Record "Test Input")
+    var
+        TempTestMethodLine: Record "Test Method Line" temporary;
+        ALTestSuite: Record "AL Test Suite";
+        TestInputsManagement: Codeunit "Test Inputs Management";
+    begin
+        ALTestSuite := this.GetOrCreateALTestSuite(AITTestMethodLine);
+
+        TempTestMethodLine."Line Type" := TempTestMethodLine."Line Type"::Codeunit;
+        TempTestMethodLine."Test Codeunit" := AITTestMethodLine."Codeunit ID";
+        TempTestMethodLine."Test Suite" := AITTestMethodLine."AL Test Suite";
+        TempTestMethodLine."Data Input Group Code" := TestInput."Test Input Group Code";
+        TempTestMethodLine."Data Input" := TestInput.Code;
+        TempTestMethodLine.Insert();
+
+        TestInputsManagement.InsertTestMethodLines(TempTestMethodLine, ALTestSuite);
+    end;
+
+    internal procedure RemoveTestMethods(var AITTestMethodLine: Record "AIT Test Method Line")
+    begin
+        this.RemoveTestMethods(AITTestMethodLine, 0, '');
+    end;
+
+    internal procedure RemoveTestMethods(var AITTestMethodLine: Record "AIT Test Method Line"; CodeunitID: Integer; DataInputName: Text[250])
+    var
+        TestMethodLine: Record "Test Method Line";
+    begin
+        TestMethodLine.SetRange("Test Suite", AITTestMethodLine."AL Test Suite");
+        if CodeunitID > 1 then
+            TestMethodLine.SetRange("Test Codeunit", CodeunitID);
+
+        if DataInputName <> '' then
+            TestMethodLine.SetRange("Data Input", DataInputName);
+
+        TestMethodLine.ReadIsolation := TestMethodLine.ReadIsolation::ReadUncommitted;
+        if TestMethodLine.IsEmpty() then
+            exit;
+
+        TestMethodLine.DeleteAll();
+        this.RemoveEmptyCodeunitTestLines(this.GetOrCreateALTestSuite(AITTestMethodLine));
+    end;
+
+    internal procedure RemoveEmptyCodeunitTestLines(ALTestSuite: Record "AL Test Suite")
+    var
+        TestMethodLine: Record "Test Method Line";
+        FunctionTestMethodLine: Record "Test Method Line";
+    begin
+        TestMethodLine.SetRange("Test Suite", ALTestSuite.Name);
+        TestMethodLine.SetRange("Line Type", TestMethodLine."Line Type"::Codeunit);
+        TestMethodLine.ReadIsolation := TestMethodLine.ReadIsolation::ReadUncommitted;
+
+        if not TestMethodLine.FindSet() then
+            exit;
+
+        repeat
```

Comment:
This entire loop could be replaced by createing a Fuction count flow field on the TestMethodLine Table. Then you can add that to the filter of the TestMethodLine record and use DeleteAll(true)

Reply:
Wont fix for now, lets keep it aligned with the existing test tool